etl/steps/data/garden/ggdc/2020-10-01/ggdc_maddison_project_database.py

```diff
@@ -0,0 +1,245 @@
+"""Load dataset for the Maddison Project Database from walden, process it, and transfer it to garden.
+
+Current dataset assumes the following approximate mappings:
+* "U.R. of Tanzania: Mainland" -> "Tanzania" (ignoring Zanzibar).
+* "Sudan (Former)" -> "Sudan" (including South Sudan).
+
+Definitions according to the Notes in the data file:
+* "gdppc": Real GDP per capita in 2011$.
+* "pop": Population, mid-year (thousands).
+
+"""
+
+import json
+from typing import cast, Dict
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from owid.walden import Catalog
+
+from etl.paths import STEP_DIR
+from etl.steps.data.converters import convert_walden_metadata
+
+# Institution name.
+NAMESPACE = "ggdc"
+# Dataset name.
+DATASET_NAME = "ggdc_maddison_project_database"
+# Original dataset publication date.
+VERSION = "2020-10-01"
+# Column name for GDP in output dataset.
+GDP_COLUMN = "GDP"
+# Column name for GDP per capita in output dataset.
+GDP_PER_CAPITA_COLUMN = "GDP per capita"
+
+
+def load_countries() -> Dict[str, str]:
+    """Load country mappings file.
+
+    Returns
+    -------
+    countries : dict
+        Country mappings.
+
+    """
+    # Define path to countries file.
+    garden_dir = STEP_DIR / "data" / "garden" / NAMESPACE / VERSION
+    files = list(garden_dir.glob("*countries.json"))
+    # Ensure there is only one countries file.
+    assert len(files) == 1
+    countries_file = files[0]
+    # Load countries from file.
+    with open(countries_file, "r") as _file:
+        countries = json.loads(_file.read())
+
+    return cast(Dict[str, str], countries)
+
+
+def load_main_data(data_file: str) -> pd.DataFrame:
+    """Load data from the main sheet of the original dataset.
+
+    Parameters
+    ----------
+    data_file : str or Path
+        Path to original data file.
+
+    Returns
+    -------
+    data : pd.DataFrame
+        Data from the main sheet of the original dataset.
+
+    """
+    # Load main sheet from original excel file.
+    data = pd.read_excel(data_file, sheet_name="Full data").rename(
+        columns={
+            "country": "Country",
+            "year": "Year",
+            "pop": "Population",
+            "gdppc": GDP_PER_CAPITA_COLUMN,
+        },
+        errors="raise",
+    )[["Country", "Year", "Population", GDP_PER_CAPITA_COLUMN]]
+    # Convert units.
+    data["Population"] = data["Population"] * 1000
+    # Create column for GDP.
+    data[GDP_COLUMN] = data[GDP_PER_CAPITA_COLUMN] * data["Population"]
+
+    # Standardize country names.
+    countries = load_countries()
+    data["Country"] = data["Country"].replace(countries)
```

Comment:
Do we need some sanity checks for unmapped / null countries or is it not worth it?

Reply:
Hi @Marigold for the time being, I've just ensured that all countries are included in the countries.json file.